src/sfdc/salesforce_bulk_endpoint.bal

```diff
@@ -174,88 +186,77 @@ public type BulkJob client object {
             JSON => {
                 if (content is json) {
                     req.setJsonPayload(content);
-                    var response = self.httpClient->post(path, req);
-                    json|ConnectorError batchResponse = checkJsonPayloadAndSetErrors(response);
-                    if (batchResponse is json){
-                        BatchInfo binfo = check BatchInfo.constructFrom(batchResponse);
-                        return binfo;
-                    } else {
-                        return batchResponse;
-                    }
                 }
-
+                if (content is string) {
+                    req.setTextPayload(content);
+                }                
                 if (content is io:ReadableByteChannel) {
-                    json payload = check convertToJson(content);
-                    req.setJsonPayload(<@untained>  payload);
-                    var response = self.httpClient->post(path, req);
-                    json|ConnectorError batchResponse = checkJsonPayloadAndSetErrors(response);
-                    if (batchResponse is json){
-                        BatchInfo binfo = check BatchInfo.constructFrom(batchResponse);
-                        return binfo;
+                    if (QUERY == self.operation) {
+                        string payload = check convertToString(content);
+                        req.setTextPayload(<@untained>  payload);
```

Comment:
```suggestion
                        req.setTextPayload(<@untainted>  payload);
```

Check all the other places as well.
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-5b542c5854666692d4661bff5f59216fR225
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-eb02955ead80dade0dfac32d254213cdR18
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-ec972bdf56dc32665948f3f1b261af47R98
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-3a4b2095243c560558aa955df684d2c5R23
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-9276837cd9f633225aedfc2e5b933789R120
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-0521f2f1fa79403950e013198aea0199R116
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-5b542c5854666692d4661bff5f59216fR196
https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/102/files#diff-5b542c5854666692d4661bff5f59216fR199